### PR TITLE
*: Support set names binary

### LINKED ISSUE #1523: tidb doesn't support SET NAMES binary
When I use mydumper(percona multithread dumper tool) to dump data from mysql, there is  "/*!40101 SET NAMES binary*/;" in schema.sql and data.sql. When I use myloader(percona multithread import tool) to import data to tidb, I enconter an error and can not import data.

### PR DESCRIPTION
**Pull request by shenli**
Fix #1523
@coocood @zimulala @tiancaiamao @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
note that this change is just grammar compatible, collation functionality related still missing...

File: util/charset/charset.go [hunk 1]
```diff
@@ -118,6 +118,9 @@ func ValidCharsetAndCollation(cs string, co string) bool {
 
 // GetDefaultCollation returns the default collation for charset
```
Comment (zimulala) [inline util/charset/charset.go]: add . at the end of a line.

**Comment by coocood** [comment]
LGTM

**Pull request merged**